lib/puppet/type/foreman_domain.rb

```diff
@@ -0,0 +1,9 @@
+require_relative '../../puppet_x/foreman/common'
+
+Puppet::Type.newtype(:foreman_domain) do
+  desc 'foreman_domain creates a domain in foreman.'
+
+  instance_eval(&PuppetX::Foreman::Common::REST_API_COMMON_PARAMS)
+  instance_eval(&PuppetX::Foreman::Common::FOREMAN_DOMAIN_PARAMS)
```

Comment:
If you're not reusing these (and it looks like you're not), it's better to inline them.

Reply:
Can you help me out here with what you're looking for?  I'm more of a C, C++, Python programmer, very new to Ruby.

Reply:
The request is to move the methods being generated by  `PuppetX::Foreman::Common::FOREMAN_DOMAIN_PARAMS` directly into this file on the `foreman_domain` type instead of the extra layer of meta-programming since there isn't another type which will reuse these methods.

Reply:
Stated another way, the methods in `lib/puppet_x/foreman/common.rb` should be moved into  `foreman_*/rest_v3.rb`.